### PR TITLE
[Wait for #2563] [ Mixed ] Create weight with var32 tensor

### PR DESCRIPTION
**Pull request by jijoongmoon**
## In this PR
This pr create the variable fp32 tensor when we create the Weight and
Optimizer Weight.

. update the manager to create Weight with  var32 tensor which
requested to weight pool.
. update the weight requests with Weight Spec and var, grad and var32
tensors which created already.
. add clone Tensor with specific type in tensor.h

Resolves:

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: nntrainer/tensor/manager.h [hunk 1]
```diff
@@ -224,7 +224,8 @@ class Manager {
    */
   std::vector<Tensor *> requestWeightOptimizerVariables(
     const std::vector<TensorDim> &dims, const std::string &name,
-    const TensorLifespan &lifespan, bool is_grad_clip, bool is_mixed_type,
+    const std::string &suffix, const TensorLifespan &lifespan,
+    bool is_grad_clip, bool is_mixed_type,
```
Comment (djeong20) [inline nntrainer/tensor/manager.h:228]: this func has incorrect description. might need to update the doc.

File: nntrainer/layers/layer_context.cpp [hunk 2]
```diff
@@ -169,6 +169,19 @@ Tensor &RunLayerContext::getWeightGrad(unsigned int idx) const {
   return weights[idx]->getGradientRef();
 }
 
+/**
+ * @brief Get the Weight Gradient tensor object
+ *
+ * @param idx Identifier of the weight
+ * @return Tensor& Reference to the weight grad tensor
```
Comment (djeong20) [inline nntrainer/layers/layer_context.cpp:176]:
```suggestion
 * @brief Get the single-precision master weight tensor object
 *
 * @param idx Identifier of the weight
 * @return Tensor& Reference to the master weight tensor
```

File: nntrainer/layers/layer_context.h [hunk 3]
```diff
@@ -463,6 +463,15 @@ class RunLayerContext {
   Tensor &getWeightGrad(unsigned int idx) const;
 
   /**
+   * @brief Get the Weight Gradient tensor object
+   *
+   * @param idx Identifier of the weight
+   * @return Tensor& Reference to the weight grad tensor
```
Comment (djeong20) [inline nntrainer/layers/layer_context.h:469]: same here
```suggestion
   * @brief Get the single-precision master weight tensor object
   *
   * @param idx Identifier of the weight
   * @return Tensor& Reference to the master weight tensor
```
```

File: nntrainer/tensor/weight.h [hunk 4]
```diff
@@ -114,6 +114,7 @@ class Weight : public Var_Grad {
    *
    * @param v Already created variable object
    * @param g Already created gradient object
+   * @param v32 Already created gradient object
```
Comment (djeong20) [inline nntrainer/tensor/weight.h:117]:
```suggestion
   * @param Already created variable32 object
```

**Review by djeong20 — APPROVED** [review]
Looks good to me except for minor fixes.

**Review by skykongkong8 — APPROVED** [review]
LGTM

**Review by DonghakPark — APPROVED** [review]
LGTM!

**Comment by jijoongmoon** [comment]
closed by  #2663

**Pull request closed without merging**